design/standard/templates/content/datatype/edit/mugoobjectrelationlist.tpl

```diff
@@ -313,7 +313,7 @@
                                 {else}
                                     <input
                                         id="field{$attribute.id}{$field_identifier}"
-                                        name="{$attribute_base}_extra_fields_{$attribute.id}[{dec($Objects.item.priority)}][{$field_identifier}]"
+                                        name="{$attribute_base}_extra_fields_{$attribute.id}[{$Relation:index}][{$field_identifier}]"
```

Comment:
Should be $Objects.index since the "section" control structure defined "var=Objects" (note it's a period separator instead of a colon in this case)